conan/tools/meson/toolchain.py

```diff
@@ -0,0 +1,272 @@
+                '14': "'c++14'", 'gnu14': "'gnu++14'",
+                '17': "'c++17'", 'gnu17': "'gnu++17'",
+                '20': "'c++1z'", 'gnu20': "'gnu++1z'"}.get(cppstd, "'none'")
+
+    @staticmethod
+    def _none_if_empty(value):
+        return "'%s'" % value if value.strip() else None
+
+    @property
+    def _context(self):
+        project_options = []
+        for k, v in self.definitions.items():
+            project_options.append("%s = %s" % (k, self._to_meson_value(v)))
+        project_options = "\n".join(project_options)
+
+        context = {
+            # https://mesonbuild.com/Machine-files.html#project-specific-options
+            "project_options": project_options,
+            # https://mesonbuild.com/Builtin-options.html#directories
+            # TODO : we don't manage paths like libdir here (yet?)
+            # https://mesonbuild.com/Machine-files.html#binaries
+            "c": self._to_meson_value(self._env.get("CC", None)),
+            "cpp": self._to_meson_value(self._env.get("CXX", None)),
+            "c_ld": self._to_meson_value(self._env.get("LD", None)),
+            "cpp_ld": self._to_meson_value(self._env.get("LD", None)),
+            "ar": self._to_meson_value(self._env.get("AR", None)),
+            "strip": self._to_meson_value(self._env.get("STRIP", None)),
+            "as": self._to_meson_value(self._env.get("AS", None)),
+            "windres": self._to_meson_value(self._env.get("WINDRES", None)),
+            "pkgconfig": self._to_meson_value(self._env.get("PKG_CONFIG", None)),
+            # https://mesonbuild.com/Builtin-options.html#core-options
+            "buildtype": self._to_meson_build_type(self._build_type) if self._build_type else None,
+            "debug": self._to_meson_value(self._debug) if self._build_type else None,
+            "default_library": self._to_meson_shared(
+                self._shared) if self._shared is not None else None,
+            # https://mesonbuild.com/Builtin-options.html#base-options
+            "b_vscrt": self._to_meson_vscrt(self._vscrt),
+            "b_staticpic": self._to_meson_value(self._fpic) if (self._shared is False and self._fpic
+                                                                is not None) else None,
+            "b_ndebug": self._to_meson_value(self._ndebug) if self._build_type else None,
+            # https://mesonbuild.com/Builtin-options.html#compiler-options
+            "cpp_std": self._to_meson_cppstd(self._cppstd) if self._cppstd else None,
+            "c_args": self._none_if_empty(self._env.get("CPPFLAGS", '') +
+                                          self._env.get("CFLAGS", '')),
+            "c_link_args": self._env.get("LDFLAGS", None),
+            "cpp_args": self._none_if_empty(self._env.get("CPPFLAGS", '') +
+                                            self._env.get("CXXFLAGS", '')),
+            "cpp_link_args": self._env.get("LDFLAGS", None),
+            "pkg_config_path": "'%s'" % os.getcwd()
```

Comment:
to be honest, I don't know if it's correct, and how to do it idiomatically
here I assume `.pc` files from `pkg-config` generator are installed to the same location as meson toolchain output.
is it always the case?

Reply:
We can assume so atm. We are investigating if it makes sense to explicitly define the layouts and locations of things.

Reply:
seems like not: https://github.com/conan-io/conan/blob/d83046a4faa2789817f8136a6d088bf524d0e50d/conan/tools/microsoft/msbuilddeps.py#L74